bpf/lbmap/lbmap.go

```diff
@@ -173,13 +208,94 @@ func LookupRevNat(key RevNatKey) (RevNatValue, error) {
 	return revnat.Convert(), nil
 }
 
+func gcd(x, y uint16) uint16 {
+	for y != 0 {
+		x, y = y, x%y
+	}
+	return x
+}
+
+type rrState struct {
+	index int
+	cw    uint16
+}
+
+func next(state *rrState, n int, max uint16, weights []uint16) uint16 {
```

Comment:
Rename function name and add comment to the function. Also, it seems this functions is for `rrState` struct. So set it as the receiver's function `func (state *rrState) next(n int, max uint16, weights []uint16) uint16 {`